OrbitGl/Disassembler.cpp

```diff
@@ -53,27 +33,33 @@ void Disassembler::Disassemble(const void* machine_code, size_t size,
     size_t j;
 
     for (j = 0; j < count; j++) {
-      LOGF("0x%" PRIx64 ":\t%-12s %s\n", insn[j].address, insn[j].mnemonic,
-           insn[j].op_str);
-      line_to_address_.push_back(insn[j].address);
+      LogLine(absl::StrFormat("0x%llx:\t%-12s %s", insn[j].address,
+                              insn[j].mnemonic, insn[j].op_str),
+              insn[j].address);
     }
 
-    // print out the next offset, after the last insn
-    LOGF("0x%" PRIx64 ":\n", insn[j - 1].address + insn[j - 1].size);
+    // Print out the next offset, after the last instruction.
+    LogLine(absl::StrFormat("0x%llx:", insn[j - 1].address + insn[j - 1].size));
 
-    // free memory allocated by cs_disasm()
+    // Free memory allocated by cs_disasm().
     cs_free(insn, count);
   } else {
-    LOG("****************\n");
-    LOG("ERROR: Failed to disasm given code!\n");
+    LogLine("****************");
+    LogLine("ERROR: Failed to disasm given code!");
   }
 
-  LOG("\n");
-
+  LogLine("");
   cs_close(&handle);
 }
 
 uint64_t Disassembler::GetAddressAtLine(size_t line) const {
   if (line >= line_to_address_.size()) return 0;
   return line_to_address_[line];
-}
\ No newline at end of file
+}
+
+void Disassembler::LogLine(std::string line, uint64_t address) {
+  // Remove any new line character.
+  absl::StrReplaceAll(line, {{"\n", ""}});
```

Comment:
where are they coming from?

Reply:
There's a mapping between lines and addresses that needs to be maintained, if someone was to call LogLine with a string containing a new line, it would mess up that mapping, that's why it's crucial to guard against it. @florian-kuebler, am I understanding this correctly?

Reply:
Right. However, I find your current solution much cleaner.